innvestigate/analyzer/reverse_map.py

```diff
@@ -0,0 +1,335 @@
+
```

Comment:
Do you have some (design) documentation on how this all works together?

I have a rough idea on what the code is doing but IMHO it would be nice to discuss/converge on the design on a high level (also to avoid conceptual problems).